arch/arm/src/stm32/stm32_1wire.c

```diff
@@ -1273,7 +1273,6 @@ int stm32_1wireuninitialize(struct onewire_dev_s *dev)
   if (--priv->refs)
     {
       nxmutex_unlock(&priv->lock);
-      kmm_free(priv);
```

Comment:
The above cod access cref without lock, but here with lock. I think we need to fix that to access lock protected only

Reply:
call stm32_1wireuninitialize with refs equal zeros is a fatal error, nxmutex_lock just make the thing more worse:

1. Modify the memory owned by other or manager
2. May hang infinitely